server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -192,6 +192,8 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.builders = Collections.unmodifiableMap(builders);
 
         threadContext = new ThreadContext(settings);
+        // adds the context to the DeprecationLogger so that it does not need to be injected everywhere
+        DeprecationLogger.setThreadContext(threadContext);
```

Comment:
can we do this as the last thing in the ctor? otherwise we need to clean up after ourself in the case of an exception?